packages/client/src/coordinator/connection/client.ts

```diff
@@ -540,8 +540,15 @@ export class StreamClient {
       await Promise.all([
         this.tokenManager.tokenReady(),
         this.guestUserCreatePromise,
-        this.connectionIdPromise,
       ]);
+      try {
+        await this.connectionIdPromise;
```

Comment:
In this case, do we even need `connectionIdPromise`? Can we just always do `waitForHealthy` instead?

Reply:
good point here, I think we dont need it too, I ll remove it

Reply:
@myandrienko not really for anonymous users we have this code inside the client

```
    // some endpoints require a connection_id to be resolved.
    // as anonymous users aren't allowed to open WS connections, we just
    // resolve the connection_id here.
    this.resolveConnectionId?.();
```

so need to keep this